app/models/piece.rb

```diff
@@ -1,8 +1,105 @@
-class Piece < ApplicationRecord
-  belongs_to :user
-  belongs_to :game
-
-  def self.types
-    %w(King Queen Rook Bishop Knight Pawn)
-  end
-end
+class Piece < ApplicationRecord
+  belongs_to :user
+  belongs_to :game
+
+  def self.types
+    %w[King Queen Rook Bishop Knight Pawn]
+  end
+
+  def obstructed?(destination_x, destination_y)
+    @destination_x = destination_x
+    @destination_y = destination_y
+
+    destination_tests = [horizontal?, vertically?, diagonal_up_right_down_left?, diagonal_up_left_down_right?]
+
+    if destination_tests.include?(true)
+      return true
+    elsif invalid_input?(destination_tests)
+      return 'Error: Invalid Input'
+    else
+      false
+    end
+  end
+
+  def invalid_input?(destination_tests)
+    invalid_input = @destination_y > 7 || @destination_x > 7 || @destination_y < 0 || @destination_x < 0
+    return true if (!destination_tests.include?(true) && !destination_tests.include?(false)) || invalid_input
+  end
+
+
+  def horizontal?

```

Comment:
@mscoutermarsh Ok I will have a look. I implemented the changes. But feels like line 10 in the Piece model becomes very long now (that is actually why I created the instance variables in the first place to avoid passing on the variables to every method and creating very long lines of code). However, I also understand that we should avoid the instance variables, so what is more important ? :-)